Mathlib/Data/Matrix/Rank.lean

```diff
@@ -274,4 +275,35 @@ theorem rank_eq_finrank_span_row [LinearOrderedField R] [Finite m] (A : Matrix m
   rw [← rank_transpose, rank_eq_finrank_span_cols, transpose_transpose]
 #align matrix.rank_eq_finrank_span_row Matrix.rank_eq_finrank_span_row
 
+section MulNonsingInv
+
+/-! ### Lemmas about pre or post multiplying by an invertible matrix. -/
+
+/-- Pre (Left) multiplying by an invertible matrix does not change the rank -/
+lemma rank_mul_isUnit [DecidableEq n] [CommRing R]
+    (A : Matrix n n R) (B : Matrix m n R) (hA : IsUnit A.det) :
+    (B ⬝ A).rank = B.rank := by
+  have hlT : LinearMap.range (mulVecLin A) = ⊤ := by
+    rw [LinearMap.range_eq_top]
+    intro x
+    use ((A⁻¹).mulVecLin x)
+    rw [mulVecLin_apply, mulVecLin_apply, mulVec_mulVec, mul_nonsing_inv _ hA, one_mulVec]
+  rw [Matrix.rank, mulVecLin_mul, LinearMap.range_comp_of_range_eq_top _ hlT, ←Matrix.rank]
+
+/-- Post (right) multiplying by an invertible matrix does not change the rank -/
+lemma rank_isUnit_mul [DecidableEq m] [Field R]
+    (A : Matrix m m R) (B : Matrix m n R) (hA : IsUnit A.det) :
+    (A ⬝ B).rank = B.rank := by
+  suffices h: LinearMap.ker ((A⬝B).mulVecLin) = LinearMap.ker (B.mulVecLin)
+  · have hB := LinearMap.finrank_range_add_finrank_ker (B.mulVecLin)
+    rw [← LinearMap.finrank_range_add_finrank_ker ((A⬝B).mulVecLin), h, add_left_inj] at hB
+    rw [Matrix.rank, Matrix.rank, hB]
+  · rw [mulVecLin_mul, LinearMap.ker_comp_of_ker_eq_bot]
+    simp_rw [LinearMap.ker_eq_bot, Function.Injective, mulVecLin_apply]
+    intros x y h
+    apply_fun Matrix.mulVec (A⁻¹) at h
+    simpa [mulVec_mulVec, nonsing_inv_mul A hA, one_mulVec] using h
```

Comment:
Usually we only indent the first goal after `suffices`
```suggestion
  rw [mulVecLin_mul, LinearMap.ker_comp_of_ker_eq_bot]
  simp_rw [LinearMap.ker_eq_bot, Function.Injective, mulVecLin_apply]
  intros x y h
  apply_fun Matrix.mulVec (A⁻¹) at h
  simpa [mulVec_mulVec, nonsing_inv_mul A hA, one_mulVec] using h
```

Reply:
I understood @semorrison comment
"Please use · and indenting to delimit the two separate goals after the suffices." to mean both goals must be indented. Was that a misunderstanding?

Reply:
I'll let @semorrison decide.